types/address.go

```diff
@@ -390,95 +306,64 @@ func (va *ValAddress) Unmarshal(data []byte) error {
 	return nil
 }
 
-// MarshalJSON marshals to JSON using Bech32.
+// MarshalJSON marshals to JSON using hex.
 func (va ValAddress) MarshalJSON() ([]byte, error) {
 	return json.Marshal(va.String())
 }
 
-// MarshalYAML marshals to YAML using Bech32.
+// MarshalYAML marshals to YAML using hex.
 func (va ValAddress) MarshalYAML() (interface{}, error) {
 	return va.String(), nil
 }
 
-// UnmarshalJSON unmarshals from JSON assuming Bech32 encoding.
+// UnmarshalJSON unmarshals from JSON assuming hex encoding.
 func (va *ValAddress) UnmarshalJSON(data []byte) error {
 	var s string
-
-	err := json.Unmarshal(data, &s)
-	if err != nil {
+	if err := json.Unmarshal(data, &s); err != nil {
 		return err
 	}
-	if s == "" {
-		*va = ValAddress{}
-		return nil
-	}
 
-	va2, err := ValAddressFromBech32(s)
-	if err != nil {
-		return err
-	}
+	*va = common.HexToAddress(s).Bytes()
 
-	*va = va2
 	return nil
 }
 
-// UnmarshalYAML unmarshals from YAML assuming Bech32 encoding.
+// UnmarshalYAML unmarshals from YAML assuming hex encoding.
 func (va *ValAddress) UnmarshalYAML(data []byte) error {
 	var s string
-
-	err := yaml.Unmarshal(data, &s)
-	if err != nil {
+	if err := yaml.Unmarshal(data, &s); err != nil {
 		return err
 	}
-	if s == "" {
-		*va = ValAddress{}
-		return nil
-	}
 
-	va2, err := ValAddressFromBech32(s)
-	if err != nil {
-		return err
-	}
+	*va = common.HexToAddress(s).Bytes()
 
-	*va = va2
 	return nil
 }
 
 // Bytes returns the raw address bytes.
 func (va ValAddress) Bytes() []byte {
-	return va
+	return va[:]
 }
 
 // String implements the Stringer interface.
 func (va ValAddress) String() string {
 	if va.Empty() {
 		return ""
 	}
-
-	key := conv.UnsafeBytesToStr(va)
-
-	if IsAddrCacheEnabled() {
-		valAddrMu.Lock()
-		defer valAddrMu.Unlock()
-
-		addr, ok := valAddrCache.Get(key)
-		if ok {
-			return addr.(string)
-		}
-	}
-	return cacheBech32Addr(GetConfig().GetBech32ValidatorAddrPrefix(), va, valAddrCache, key)
+	// TODO HV2: the "0x" addition is preserved from heimdall and necessary to have eth addresses string representations
+	//  Does it need to be done for all `Tx` related types? Like `TxHash`
+	return "0x" + common.Bytes2Hex(va.Bytes())
```

Comment:
TxHash is a different representation from an AccountAddress so no, you wouldn't prepend the `0x` prefix to a TxHash